src/Aspire.Hosting/Publishing/IContainerRuntime.cs

```diff
@@ -0,0 +1,9 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Aspire.Hosting.Publishing;
+
+internal interface IContainerRuntime
+{
+    public Task<string> BuildImageAsync(string contextPath, string dockerfilePath, string imageName, CancellationToken cancellationToken);
```

Comment:
These needs to take arbitrary arguments.

Reply:
Next PR will tackle this for things like build args etc.